addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1672,6 +1632,24 @@ options.registry.anchorName = options.Class.extend({
         }
         this.$target.trigger('content_changed');
     },
+    /**
+     * Returns anchor text.
+     *
+     * @private
+     */
+     _getAnchor: function (previewMode, value, $opt) {
+         if (!this.$target[0].id) {
+             const $titles = this.$target.find('h1, h2, h3, h4, h5, h6');
+             const title = $titles.length > 0 ? $titles[0].innerText : this.data.snippetName;
+             const anchorName = encodeURIComponent(title);
+             let n = '';
+             while (document.getElementById(anchorName + n)) {
+                 n = (n || 1) + 1;
+             }
+             this._setAnchorName(anchorName + n);
+         }
+         return '#' + this.$target[0].id;
```

Comment:
In fact I missed that part in my first review but the spec is to actually copy a DOM link so a `<a href="..." ...`